modules/ts/src/ts_func.cpp

```diff
@@ -515,7 +515,7 @@ void extract(const Mat& src, Mat& dst, int coi)
 void transpose(const Mat& src, Mat& dst)
 {
     CV_Assert(src.data != dst.data && "Inplace is not support in cvtest::transpose");
-    CV_Assert(src.dims == 2);
+    CV_Assert(src.dims <= 2);
```

Comment:
`transpose()` for 1D/0D should be prohibited to avoid coding mistakes.

User should reshape mat to 2D first in explicit way.